src/compiler/parser.ts

```diff
@@ -3015,7 +3015,11 @@ namespace ts {
                 let newPrecedence = getBinaryOperatorPrecedence();
 
                 // Check the precedence to see if we should "take" this operator
-                if (newPrecedence <= precedence) {
+                if (token === SyntaxKind.AsteriskAsteriskToken && newPrecedence < precedence) {
+                    // ** operator is right-assocative
+                    break;
+                }
+                else if (token !== SyntaxKind.AsteriskAsteriskToken && newPrecedence <= precedence) {

```

Comment:
Sorry, as we discusssed offline, this should be `token !== SyntaxKind.AsteriskAsteriskToken || newPrecedence < precedence`